esi_leap/objects/flocx_market_client.py

```diff
@@ -63,3 +64,65 @@ def send_offer(self, data):
                 r.status_code,
                 r.text)
         return r.status_code
+
+    def get_contracts(self):
+        url = CONF.flocx_market.endpoint_override + \
+        '/offer_contract_relationship?status=unretrieved'
+        r = self.get(url)
+        if r.status_code != 200:
+            LOG.warning(
+                "Failed to retrieve contracts from flocx-market. Got HTTP %s: %s",
+                r.status_code,
+                r.text)
+        if r.content is not None:
+            contracts = []
+            contract_offer_table = {}
+            contract_offer_list = json.loads(r.content)
+            # create contract offers hash table
+            # key: contract_id, value: (offer_id, offer_contract_id) list
+            for c_o in contract_offer_list:
+                if not contract_offer_table.get(str(c_o["contract_id"])):
+                    o_c_r_tuple = (str(c_o["marketplace_offer_id"]), str(c_o["offer_contract_relationship_id"]))
+                    contract_offer_table[str(c_o["contract_id"])] = [o_c_r_tuple]
+                else:
+                    contract_offer_table[str(c_o["contract_id"])].append(o_c_r_tuple)
+
+            # retrieve contracts and offers data from flocx-market iteratively
+            for contract_id, o_c_rs in contract_offer_table.items():
+                contract_data = {}
+                url1 = CONF.flocx_market.endpoint_override + '/contract/' + contract_id
+                r_contract = self.get(url1)
+                c = json.loads(r_contract.content)
+                contract_data["contract_id"] = str(c["contract_id"])
+                # contract_data["time_created"] = str(c["time_created"])
+                contract_data["start_time"] = str(c["start_time"])
+                contract_data["end_time"] = str(c["end_time"])
+                contract_data["status"] = str(c["status"])
+                contract_data["cost"] = str(c["cost"])
+                contract_data["bid_id"] = str(c["bid_id"])
```

Comment:
Why are we calling `str()` on all of these attributes? If this transformation is necessary, it should probably be in a loop rather than a series of individual statements.

Reply:
I was trying to modify the string data got rom json.load() from unicode to utf-8. Now I add a object_hook to do the convert. https://github.com/DanNiESh/esi-leap/blob/cd637868eaecdb30fc59538a34a1a0bb4ea78a04/esi_leap/common/flocx_market.py#L12